xds/csds/csds.go

```diff
@@ -76,11 +75,12 @@ type ClientStatusDiscoveryServer struct {
 func NewClientStatusDiscoveryServer() (*ClientStatusDiscoveryServer, error) {
 	xdsC, err := newXDSClient()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create xds client: %v", err)
+		logger.Warningf("failed to create xds client: %v", err)
+		// Return an explicit nil here, otherwise the nil returned from
+		// client.New() is a typed nil (type *Client, not the interface).
+		return &ClientStatusDiscoveryServer{}, nil
```

Comment:
It took me a long time to figure out what this comment means.  Let's change it:

```go
// xdsC is a typed nil here, so we if we assign it to the xdsClient field, the
// nil checks in the handlers will not handle it properly.
```

Or you could fix this by doing:

```go
newXDSClient = func() (xdsClientInterface, error) {
	c, err := client.New()
	if err != nil {
		return nil, err
	}
	return c, nil
}
```

Or slightly better:

```go
newXDSClient = func() xdsClientInterface {
	c, err := client.New()
	if err != nil {
		logger.Warningf(...)
		return nil
	}
	return c
}
```


But _ACTUALLY_, why do we even have `newXDSClient`?  For testing, why not construct a `ClientStatusDiscoverServer` manually?

Reply:
Updated to `newXDSClient = func() xdsClientInterface`

> But ACTUALLY, why do we even have newXDSClient? For testing, why not construct a ClientStatusDiscoverServer manually?

Why is this better? If we add more things to `NewClientStatusDiscoveryServer()`, they won't be covered.